packages/global-styles/src/renderer.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useLayoutEffect } from '@wordpress/element';
+
+export function useRenderedGlobalStyles( styles = {} ) {
+	useGlobalStylesEnvironment();
+	const generatedStyles = compileStyles( styles );
+
+	useEffect( () => {
+		// styleNodeId should match server ID, see global-styles.php
+		const styleNodeId = 'global-styles-inline-css';
+		let styleNode = document.getElementById( styleNodeId );
+
+		if ( ! styleNode ) {
+			styleNode = document.createElement( 'style' );
+			styleNode.id = styleNodeId;
+			document
+				.getElementsByTagName( 'head' )[ 0 ]
+				.appendChild( styleNode );
+		}
+
+		styleNode.innerText = generatedStyles;
+	}, [ generatedStyles ] );
+}
+
+function useGlobalStylesEnvironment() {
```

Comment:
Would it be possible to add the wp-gs class directly on the server if global styles are supported?

Reply:
We're no longer relying on the `wp-gs` so I've removed the code that dealt with that.